### PR TITLE
AI Agents Guide

### PR DESCRIPTION
**Pull request by oandreeva-nv**
This PR adds a dedicated AI Agents Guide section to our tutorials. 

Initial entries:
 - Constrained decoding: #109 
 - Function Calling: #113

### CODE REVIEW

File: AI_Agents_Guide/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+for users who are already familiar with the basics of Triton and are looking to
+expand their knowledge.
+
+For beginners, we recommend starting with the
+[Conceptual Guide](tutorials/Conceptual_Guide/README.md), which covers
+foundational concepts and basic setup of Triton Inference Server.
+
+## AI agents and Agentic Workflows
+
+Modern large language models (LLMs) are integral components of AI agents —
```
Comment (statiraju) [inline AI_Agents_Guide/README.md:43]: are there examples of AI agents which integrate LLMs? wondering if some high level understanding of AI agents would be good to add.

Comment (oandreeva-nv) [reply AI_Agents_Guide/README.md:43]: you mean real-world examples, or description, or both?

Comment (statiraju) [reply AI_Agents_Guide/README.md:43]: both. but can happen as an additional work later on. approving for merge for first version of documentation.

**Review by statiraju — DISMISSED** [review]

File: AI_Agents_Guide/README.md (hunk 1 above)
Comment (oandreeva-nv) [reply AI_Agents_Guide/README.md:43]: Created ticket: DLIS-7291

File: AI_Agents_Guide/Constrained_Decoding/artifacts/client.py [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+        FLAGS.streaming,
+        FLAGS.beam_width,
+        FLAGS.overwrite_output_text,
+        return_context_logits_data,
+        return_generation_logits_data,
+        FLAGS.end_id,
+        FLAGS.pad_id,
+        FLAGS.verbose,
+        logits_post_processor_name=FLAGS.logits_post_processor_name,
+    )
```
[the remark this answers is not in the document]
Comment (oandreeva-nv) [reply AI_Agents_Guide/Constrained_Decoding/artifacts/client.py:276]: the linked function is defined in the script, belonging to the other tutorial

**Review by statiraju — APPROVED** [review]

**Pull request merged**